Profiler/config/ConfigParser.h

```diff
@@ -5,10 +5,13 @@
 #include "utils/StringUtils.h"
 #include "yaml-cpp/yaml.h"
 #include "utils/Testing.h"
-
 /** A process-specific config section. */
 struct ProcessSection {
-	std::regex processRegex;
+	/** Regular expression that must match the full process path for the section to be applied. This field is always set. */
```

Comment:
`This field defaults to ".*"`? I think it is important to know that if the field is not set explicitly, the regex it contains matches everything, because otherwise the way you implemented `sectionMatches()` wouldn't work.